### PR TITLE
submit extra data from CRON_SENTRY_ env variables

### PR DESCRIPTION
**Pull request by filippog**

### CODE REVIEW

File: cron_sentry/runner.py [hunk 1]
```diff
@@ -140,19 +154,22 @@ def report_fail(self, exit_status, last_lines_stdout, last_lines_stderr, elapsed
 
         message = "Command \"%s\" failed" % (self.command,)
 
-        client = Client(transport=HTTPTransport, dsn=self.dsn, string_max_length=self.string_max_length)
+        client = Client(transport=HTTPTransport, dsn=self.dsn,
+                string_max_length=self.string_max_length, verify_ssl=0)
```
Comment (hltbra) [inline cron_sentry/runner.py]: Why change `verify_ssl`? Debugging only, I guess?!

Comment (hltbra) [inline cron_sentry/runner.py]: Why change `verify_ssl`? Debugging only, I guess?!

Comment (filippog) [inline cron_sentry/runner.py]: doh! thanks, yeah debugging only, removed!

**Comment by hltbra** [comment]
Hey @filippog, sorry for the delay. I am still thinking about this approach. I would like to have command-line options too.

The current approach assumes there will be multiple env vars, instead of only one. I am not sure this is the best approach, maybe a single variable would be best.

What do you think, @filippog?

**Comment by filippog** [comment]
hey @hltbra no problem! yeah I agree command line options are nice to have too. I take it command line options would override env vars?
I assumed multiple env variables since `extra` is a mapping, I think a single env variable would work too but then it would be limited to passing only one key=value pair?
For example my use case for multiple variables would be something like this:

```
CRON_SENTRY_JOB=myjob
CRON_SENTRY_STAGE=canary
cron-sentry /usr/local/bin/myjob
```

what do you think?

**Comment by asfaltboy** [comment]
Hi guys, great project! I think we could support both args and env vars, [à la click][1].

[1]: http://click.pocoo.org/5/arguments/#environment-variables

**Comment by filippog** [comment]
hi @asfaltboy yep that would work for me (env + command line), what do you think @hltbra ?

**Comment by filippog** [comment]
hi @hltbra ! Thoughts on supporting env variables and command line like @asfaltboy suggested? thanks!

File: cron_sentry/runner.py [hunk 2]
```diff
@@ -76,6 +76,16 @@ def update_dsn(opts):
             return
 
 
+def _extra_from_env(env):
+    res = {}
+    for k, v in iter(env.items()):
+        if k.startswith('CRON_SENTRY_'):
+            extra_key = k[12:]
```
Comment (hltbra) [inline cron_sentry/runner.py]: I think the following is better long term:
```python
env_var_prefix = 'CRON_SENTRY_'
if k.startswith(env_var_prefix):
    extra_key = k[len(env_var_prefix):]
```

**Comment by hltbra** [comment]
@filippog: what @asfaltboy suggested is something different from this PR. This PR is meant to add extra parameters to be sent to Sentry instead of using cron-sentry via CLI or env vars, such as:
```bash
cron-sentry --quiet
export CRON_SENTRY_QUIET=true
cron-sentry
```

I like the idea of having both extra commands and also specifying options via env vars. Would you change this PR to use the `CRON_SENTRY_EXTRA_` prefix, @filippog? Then I might add a command line option `--extra` (accumulative) so we can have both ways of specifying it (`cron-sentry --extra job=myjob --extra stage=canary` or `--extra job=myjob;stage=canary`)

File: cron_sentry/runner.py [hunk 3]
```diff
@@ -76,6 +76,16 @@ def update_dsn(opts):
             return
 
 
+def _extra_from_env(env):
+    res = {}
+    for k, v in iter(env.items()):
```
Comment (filippog) [inline cron_sentry/runner.py]: Indeed, fixed!

File: cron_sentry/runner.py (hunk 2 above)
Comment (filippog) [inline cron_sentry/runner.py]: Agreed, fixed too!

**Comment by filippog** [comment]
> I like the idea of having both extra commands and also specifying options via env vars. Would you change this PR to use the CRON_SENTRY_EXTRA_ prefix, @filippog? Then I might add a command line option --extra (accumulative) so we can have both ways of specifying it (cron-sentry --extra job=myjob --extra stage=canary or --extra job=myjob;stage=canary)

Sure! `CRON_SENTRY_EXTRA_` sounds good too.

**Comment by filippog** [comment]
One test fails on py3 but it doesn't seem related to this change. I've rebased on current master.

**Comment by hltbra** [comment]
@filippog: I have fixed the tests (master branch) and merged your PR (rebased on top of master). 

I will update the CHANGELOG and release a new version. Thanks!

**Comment by hltbra** [comment]
@filippog: Release 0.7.0 includes this PR. Thanks again.

**Comment by filippog** [comment]
awesome @hltbra thanks a lot!

**Pull request merged**